src/components/NotificationModal.tsx

```diff
@@ -0,0 +1,200 @@
+import React from "react";
+import { Modal, Select, InputNumber, Input } from "antd";
+import { uiColors } from "@leafygreen-ui/palette";
+import styled from "@emotion/styled";
+import { Button } from "components/Button";
+import { H2, Body } from "@leafygreen-ui/typography";
+import get from "lodash/get";
+
+import set from "lodash/set";
+import { SubscriptionMethod } from "types/subscription";
+import { v4 as uuid } from "uuid";
+import {
+  useNotificationModal,
+  UseNotificationModalProps,
+} from "hooks/useNotificationModal";
+const { Option } = Select;
+
+interface ModalProps extends UseNotificationModalProps {
+  visible: boolean;
+  subscriptionMethods: SubscriptionMethod[];
+}
+
+export const NotificationModal: React.FC<ModalProps> = ({
+  visible,
+  onCancel,
+  subscriptionMethods,
+  subscriptionMethodControls,
+  triggers,
+  resourceId,
+  resourceType,
+}) => {
+  const {
+    selectedSubscriptionMethod,
+    onClickSave,
+    mutationLoading,
+    isFormValid,
+    selectedTriggerId,
+    setSelectedTriggerId,
+    extraFields,
+    extraFieldInputVals,
+    setExtraFieldInputVals,
+    setSelectedSubscriptionMethod,
+    target,
+    setTarget,
+    extraFieldErrorMessages,
+  } = useNotificationModal({
+    subscriptionMethodControls,
+    triggers,
+    resourceType,
+    resourceId,
+    onCancel,
+  });
+
+  const currentMethodControl = subscriptionMethodControls[
+    selectedSubscriptionMethod
+  ] as SubscriptionMethodControl;
+  const label = get(currentMethodControl, "label");
+  const placeholder = get(currentMethodControl, "placeholder");
+  const targetPath = get(currentMethodControl, "targetPath");
+
+  return (
+    <Modal
+      data-test-id="subscription-modal"
+      title={<H2>Add Subscription</H2>}
+      visible={visible}
+      onCancel={onCancel}
+      footer={[
+        <Button
+          key="cancel"
+          onClick={onCancel}
+          dataCy="cancel-subscription-button"
+        >
+          Cancel
+        </Button>,
+        <Button
+          key="save"
+          dataCy="save-subscription-button"
+          loading={mutationLoading}
+          disabled={!isFormValid}
+          onClick={onClickSave}
+          variant="danger"
+        >
+          Save
+        </Button>,
+      ]}
+      width="50%"
+      wrapProps={{
+        "data-cy": "task-notification-modal",
+      }}
+    >
+      <>
+        when
+        <StyledSelect
+          value={selectedTriggerId}
+          onChange={(v: string) => {
+            setSelectedTriggerId(v);
+          }}
+          data-test-id="when-select"
+        >
+          {triggers.map((t) => (
+            <Option
+              key={t.trigger}
+              value={t.trigger}
+              data-test-id={`${t.trigger}-option`}
+            >
+              {t.label}
+            </Option>
+          ))}
+        </StyledSelect>
+      </>
+      {(extraFields ?? []).map(({ text, key }) => (
```

Comment:
you can just do `extraFields && extraFields.map(....` no point in mapping over an empty array.